experimental/packages/opentelemetry-api-metrics/src/types/Metric.ts

```diff
@@ -101,27 +101,29 @@ export type MetricAttributeValue = SpanAttributeValue;
 /**
  * The observable callback for Observable instruments.
  */
-export type ObservableCallback = (observableResult: ObservableResult) => void | Promise<void>;
+export type ObservableCallback<AttributesTypes extends MetricAttributes=MetricAttributes> =
+  (observableResult: ObservableResult<AttributesTypes>) => void | Promise<void>;
 
 /**
  * The observable callback for a batch of Observable instruments.
  */
-export type BatchObservableCallback = (observableResult: BatchObservableResult) => void | Promise<void>;
+export type BatchObservableCallback<AttributesTypes extends MetricAttributes=MetricAttributes> =
+  (observableResult: BatchObservableResult<AttributesTypes>) => void | Promise<void>;
 
-export interface Observable {
+export interface Observable<AttributesTypes extends MetricAttributes=MetricAttributes> {
   /**
    * Sets up a function that will be called whenever a metric collection is initiated.
    *
    * If the function is already in the list of callbacks for this Observable, the function is not added a second time.
    */
-  addCallback(callback: ObservableCallback): void;
+  addCallback(callback: ObservableCallback<AttributesTypes>): void;
 
   /**
    * Removes a callback previously registered with {@link Observable.addCallback}.
    */
-  removeCallback(callback: ObservableCallback): void;
+  removeCallback(callback: ObservableCallback<AttributesTypes>): void;
 }
 
-export type ObservableCounter = Observable;
-export type ObservableUpDownCounter = Observable;
-export type ObservableGauge = Observable;
+export type ObservableCounter<AttributesTypes extends MetricAttributes=MetricAttributes> = Observable<AttributesTypes>;
+export type ObservableUpDownCounter<AttributesTypes extends MetricAttributes=MetricAttributes> = Observable<AttributesTypes>;
+export type ObservableGauge<AttributesTypes extends MetricAttributes=MetricAttributes> = Observable<AttributesTypes>;
```

Comment:
Would you mind adding a space between the `=` sign in type parameter declarations in this files too?